internal/aws/cluster/update.go

```diff
@@ -3,58 +3,70 @@ package cluster
+	var nextToken *string
+	var asgOutput *autoscaling.DescribeAutoScalingGroupsOutput
+	var launchTemplateVersionsOutput *ec2.DescribeLaunchTemplateVersionsOutput
 
-	hostGroupParams = common.HostGroupParams{
-		SecurityGroupsIds: launchTemplateData.SecurityGroupIds,
-		ImageID:           *launchTemplateData.ImageId,
-		KeyName:           *launchTemplateData.KeyName,
-		IamArn:            *launchTemplateData.IamInstanceProfile.Arn,
-		InstanceType:      *launchTemplateData.InstanceType,
-		Subnet:            *launchTemplateData.NetworkInterfaces[0].SubnetId,
-		VolumeName:        *launchTemplateData.BlockDeviceMappings[0].DeviceName,
-		VolumeType:        *launchTemplateData.BlockDeviceMappings[0].Ebs.VolumeType,
-		VolumeSize:        *launchTemplateData.BlockDeviceMappings[0].Ebs.VolumeSize,
-		MaxSize:           maxSize,
+	for asgOutput == nil || nextToken != nil {
+		asgOutput, err = svcAsg.DescribeAutoScalingGroups(
+			&autoscaling.DescribeAutoScalingGroupsInput{
+				NextToken: nextToken,
+			},
+		)
+		if err != nil {
+			return
+		}
+		for _, asg := range asgOutput.AutoScalingGroups {
+			asgRole := autoscaling2.GetAsgTagValue(asg, RoleTagKey)
+			asgClusterName := autoscaling2.GetAsgTagValue(asg, cluster.ClusterNameTagKey)
+			if asgRole != string(role) || asgClusterName != string(clusterName) {
+				continue
+			}
+			log.Debug().Msgf("Generating %s host group params ...", *asg.AutoScalingGroupName)
+			launchTemplateVersionsOutput, err = svcEc2.DescribeLaunchTemplateVersions(&ec2.DescribeLaunchTemplateVersionsInput{
+				LaunchTemplateName: asg.LaunchTemplate.LaunchTemplateName,
+			})
+			if err != nil {
+				return
+			}
+			launchTemplateData := launchTemplateVersionsOutput.LaunchTemplateVersions[0].LaunchTemplateData
+			hostGroupsParams = append(hostGroupsParams,
+				common.HostGroupParams{
+					SecurityGroupsIds: launchTemplateData.SecurityGroupIds,
+					ImageID:           *launchTemplateData.ImageId,
+					KeyName:           *launchTemplateData.KeyName,
+					IamArn:            *launchTemplateData.IamInstanceProfile.Arn,
+					InstanceType:      *launchTemplateData.InstanceType,
+					Subnet:            *launchTemplateData.NetworkInterfaces[0].SubnetId,
+					VolumeName:        *launchTemplateData.BlockDeviceMappings[0].DeviceName,
+					VolumeType:        *launchTemplateData.BlockDeviceMappings[0].Ebs.VolumeType,
+					VolumeSize:        *launchTemplateData.BlockDeviceMappings[0].Ebs.VolumeSize,
+					MaxSize:           *asg.MaxSize,
+				})
+		}
+		nextToken = asgOutput.NextToken
 	}
-
 	return
 }
 
-func generateHostGroupFromLaunchTemplate(clusterName cluster.ClusterName, role common.InstanceRole, name common.HostGroupName) (hostGroup HostGroup, err error) {
-	hostGroupParams, err := fetchClusterLaunchTemplateParams(clusterName, name)
+func generateHostGroupFromLaunchTemplate(clusterName cluster.ClusterName, role common.InstanceRole, name common.HostGroupName) (hostGroups []HostGroup, err error) {
```

Comment:
ok, function name is not right now and that's what confused me in above comment.
it is basically GetHostGroupsByRole